internal/controller/controlplane/helper.go

```diff
@@ -277,45 +278,37 @@ func (c *K0sController) createAutopilotPlan(ctx context.Context, kcp *cpv1beta1.
 
 	timestamp := fmt.Sprintf("%d", time.Now().Unix())
 	plan := []byte(`
-	{
-		"apiVersion": "autopilot.k0sproject.io/v1beta2",
-		"kind": "Plan",
-		"metadata": {
-		  "name": "autopilot"
-		},
-		"spec": {
-			"id": "id-` + kcp.Name + `-` + timestamp + `",
-			"timestamp": "` + timestamp + `",
-			"commands": [{
-				"k0supdate": {
-					"version": "` + kcp.Spec.Version + `",
-					"platforms": {
-						"linux-amd64": {
-							"url": "` + amd64DownloadURL + `"
-						},
-						"linux-arm64": {
-							"url": "` + arm64DownloadURL + `"
-						},
-						"linux-arm": {
-							"url": "` + armDownloadURL + `"
-						}
-					},
-					"targets": {
-						"controllers": {
-							"discovery": {
-							    "static": {
-									"nodes": ["` + strings.Join(machines.Names(), `","`) + `"]
-								}
-							}
-						}
-					}
-				}
-			}]
-		}
-	}`)
-
-	return clientset.RESTClient().Post().
+apiVersion: autopilot.k0sproject.io/v1beta2
```

Comment:
Switching to YAML since `Patch(types.ApplyPatchType)` sets content-type header to `application/apply-patch+yaml`